commit 8007365a025fd80f682408098ffbc8e5bfa6473f (refs/changes/14/2514/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-09-08T10:47:38-07:00 (2 years, 1 month ago)
    
    IMGAPI-643 docker pull crashes if an unexpected x-registry-config header is used
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/CHANGES.md b/CHANGES.md
index ec9f7fb..06a0bdc 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,12 +1,18 @@
 # IMGAPI changelog
 
+## 4.0.6
+
+- IMGAPI-643 docker pull crashes if an unexpected x-registry-config header is
+  used
+
 ## 4.0.5
 
 - DOCKER-1097 docker pull fails for registry server that has no auth setup
 
 ## 4.0.4
 
-- joyent/sdc-imgapi#13 imgapi crash when pulling from docker.io with invalid credentials
+- joyent/sdc-imgapi#13 imgapi crash when pulling from docker.io with invalid
+  credentials
 
 ## 4.0.3
 
diff --git a/lib/docker.js b/lib/docker.js
index 5394a3b..9a58578 100644
--- a/lib/docker.js
+++ b/lib/docker.js
@@ -1478,14 +1478,16 @@ function adminImportDockerImage(opts, callback) {
         // Find registry auth from the registry hostname map. Note that Docker
         // uses a special legacy name for the "official" docker registry, so
         // check for that too.
-        var dockerV1CompatName = 'https://index.docker.io/v1/';
-        var indexName = rat.index.name;
-        if (regConfig.hasOwnProperty(indexName)) {
-            regAuth = regConfig[indexName];
-        } else if (indexName === 'docker.io' &&
-            regConfig.hasOwnProperty(dockerV1CompatName))
-        {
-            regAuth = regConfig[dockerV1CompatName];
+        if (regConfig) {
+            var dockerV1CompatName = 'https://index.docker.io/v1/';
+            var indexName = rat.index.name;
+            if (regConfig.hasOwnProperty(indexName)) {
+                regAuth = regConfig[indexName];
+            } else if (indexName === 'docker.io' &&
+                regConfig.hasOwnProperty(dockerV1CompatName))
+            {
+                regAuth = regConfig[dockerV1CompatName];
+            }
         }
     }
     if (req.headers['x-registry-auth']) {
diff --git a/package.json b/package.json
index bfdd7f8..13e784e 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgapi",
   "description": "Image API to manage images for SDC 7",
-  "version": "4.0.5",
+  "version": "4.0.6",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
